extensions/content-manager/admin/src/containers/EditView/index.js

```diff
@@ -0,0 +1,317 @@
+import React, { memo, useCallback, useMemo, useEffect, useReducer, useRef, useState } from 'react';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { useHistory, useLocation, useRouteMatch } from 'react-router-dom';
+import { BackHeader, LiLink, request } from 'strapi-helper-plugin';
+import pluginId from '../../pluginId';
+import Container from '../../components/Container';
+import DynamicZone from '../../components/DynamicZone';
+import FormWrapper from '../../components/FormWrapper';
+import FieldComponent from '../../components/FieldComponent';
+import Inputs from '../../components/Inputs';
+import SelectWrapper from '../../components/SelectWrapper';
+import getInjectedComponents from '../../utils/getComponents';
+import EditViewDataManagerProvider from '../EditViewDataManagerProvider';
+import EditViewProvider from '../EditViewProvider';
+import Header from './Header';
+import createAttributesLayout from './utils/createAttributesLayout';
+import { LinkWrapper, SubWrapper } from './components';
+import init from './init';
+import reducer, { initialState } from './reducer';
+
+/* eslint-disable  react/no-array-index-key */
+
+const EditView = ({ components, currentEnvironment, deleteLayout, layouts, plugins, slug }) => {
+
+  const { search, pathname } = useLocation();
+  const [selectedOption, setSelectedOption] = useState({})
+  const [isStructureCorrect, setIsStructureCorrect] = useState(true)
+  const collectionName = get(layouts, [slug, 'contentType', 'schema', 'collectionName']);
+  const currentContentTypeId = pathname.substring(pathname.lastIndexOf('/') + 1);
+  const isRelationViewEnabled = get(layouts, [slug, 'contentType', 'settings', 'relation']);
+  const selectedRelationName = get(layouts, [slug, 'contentType', 'settings', 'relationName']);
+
+  const getStructureCorrectness = async (collectionName, relationName, parentId, childId) => {
+    try {
+      const structureCorrectness = await request(`/content-manager/${collectionName}/validateRelation/${relationName}/${parentId}/${childId}`, {
```

Comment:
as I know Admin panel uses sagas, and at least we should do some service instead of fetching directly in component

Reply:
is it similar to strapi source code